src/Compilers/CSharp/Test/Semantic/Semantics/InteractiveSemanticModelTests.cs

```diff
@@ -268,6 +268,32 @@ public void BindQueryVariable()
             Assert.Equal(SpecialType.System_Char, semanticInfo.Type.SpecialType);
         }
 
+        [Fact]
+        public void TestLocalFunctionDeclaration()
+        {
+            var compilation = CreateCompilation("""
+                                  class Test
+                                  {
+                                    void M()
+                                    {
+                                        int LocalFunc(string s) {}
+                                    }
+                                  }
+                                  """);
+            var tree = compilation.SyntaxTrees[0];
+            var semanticModel = compilation.GetSemanticModel(tree);
+            var root = tree.GetCompilationUnitRoot();
+            var localFunction = (LocalFunctionStatementSyntax)((MethodDeclarationSyntax)((ClassDeclarationSyntax)root.Members[0]).Members[0]).Body!.Statements[0];
+            IMethodSymbol methodSymbol = semanticModel.GetDeclaredSymbol(localFunction);
+
+            Assert.NotNull(methodSymbol);
+            Assert.Equal(SpecialType.System_Int32, methodSymbol.ReturnType.SpecialType);
+            Assert.Equal("LocalFunc", methodSymbol.Name);
+            Assert.Equal(1, methodSymbol.Parameters.Length);
+            Assert.Equal(SpecialType.System_String, methodSymbol.Parameters[0].Type.SpecialType);
+            Assert.Equal("s", methodSymbol.Parameters[0].Name);
```

Comment:
```suggestion
            Assert.Equal("int LocalFunc(string s)", methodSymbol.ToTestDisplayString());
```

Not 100% certain I got that baseline string right, but it should be close.